src/main/java/de/tum/in/www1/artemis/service/ParticipationService.java

```diff
@@ -185,6 +192,30 @@ else if (exercise instanceof QuizExercise || exercise instanceof ModelingExercis
         return participation;
     }
 
+    /**
+     * Initializes a new text or modeling submission (depending on the type of the given exercise), connects it with the given participation and stores it in the database.
+     *
+     * @param participation the participation for which the submission should be initialized
+     * @param exercise      the corresponding exercise, should be either a text or modeling exercise, otherwise it will instantly return and not do anything
+     */
+    private void initializeSubmission(Participation participation, Exercise exercise) {
+        if (exercise instanceof ProgrammingExercise || exercise instanceof QuizExercise) {
+            return;
+        }
+
+        Submission submission;
+        if (exercise instanceof ModelingExercise) {
+            submission = new ModelingSubmission();
+        }
+        else {
+            submission = new TextSubmission();
```

Comment:
Please handle file upload exercises as well (even if they are not fully supported yet)

Reply:
Done.